File: back-end/src/main/java/go/travels/backend/utils/security/UserDetailsServiceImpl.java
package go.travels.backend.utils.security;

import go.travels.backend.document.User;
import go.travels.backend.repositories.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    @Autowired
    private UserRepository repository;

    @Override
    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
        User u = repository.findByEmail(email);

        if (u == null) {
            throw new UsernameNotFoundException(email);
        }

        return new UserSS(u.getId(), u.getEmail(), u.getPassword(), u.getProfile());
    }
}

File: go-travel-service/go-travel-service/src/main/kotlin/com/gotravel/domain/service/UserDetailsServiceImpl.kt
package com.gotravel.domain.service

import com.gotravel.domain.repositories.UserRepository
import com.gotravel.resources.security.UserDetailsImpl
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl(private val userRepository: UserRepository): UserDetailsService {

    override fun loadUserByUsername(email: String?): UserDetails {
        val user = userRepository.findByEmail(email) ?: throw UsernameNotFoundException(email)

        return UserDetailsImpl(user)
    }
}